doc/src/cylc-user-guide/siterc.tex

```diff
@@ -748,6 +753,142 @@ \subsection{[hosts]}
+
+Configure thresholds for excluding insufficient hosts and a method for
+ranking the remaining hosts to be applied in selection of the most suitable
+\lstinline=run host=, from those configured, at start-up whenever a set host
+is not specified on the command line via the \lstinline@--host=@ option.
+
+\paragraph[rank]{[suite servers] \textrightarrow [[run host select]] \textrightarrow rank}
+
+The method to use to rank the \lstinline=run host= list in order of
+suitability.
+
+\begin{myitemize}
+\item {\em type:} string (which must be one of the options outlined below)
+\item {\em default:} \lstinline=random=
+\item {\em options:}
+    \begin{myitemize}
+    \item {\bf random} - rank for suitability and select a host at random
+    \item {\bf load:1} - rank for suitability and select the host with the
+lowest load average over 1 minute, in KB, (as given by the
+\lstinline=uptime= command)
+    \item {\bf load:5} - as for \lstinline=load:1= above, but over 5 minutes
+    \item {\bf load:15} - as for \lstinline=load:1= above, but over 15 minutes
+    \item {\bf memory} - rank for suitability and select the host with the
+highest usable memory, in KB, as specified under \lstinline=/proc/meminfo=
+    \item {\bf disk-space:PATH} - rank for suitability and select the host
+with the highest free disk space for a given mount directory path
+\lstinline=PATH=, in KB, as given by the \lstinline=df= command, where
+multiple paths may be specified individually i.e. via
+\lstinline=disk-space:PATH_1= and \lstinline=disk-space:PATH_2=, etc.
+    \end{myitemize}
+\item {\em default:} (none)
+\end{myitemize}
+
+\paragraph[thresholds]{[suite servers] \textrightarrow [[run host select]] \textrightarrow thresholds}
+
+A list of metric measure and associated numeric value specifications
+against which all configured run hosts are compared; if for a given allowed
+host the extracted value of {\em any} of the provided measures does not
+meet the cutoff provided by the corresponding numeric value, which is a
+minimum or a maximum requirement depending on the measure context, the host
+is excluded from the list of run hosts to undergo ranking i.e. excluded as a
+potential run host. Usable memory (\lstinline=memory=) and free disk space
+(\lstinline=disk-space:PATH=) threshold values set a {\em minimum} value,
+which must be exceeded, whereas load average (\lstinline=load:1=,
+\lstinline=load:5= and \lstinline=load:15=) threshold values set a
+{\em maximum}, which must not be.
+
+\begin{myitemize}
+\item {\em type:} string in format
+\lstinline=<MEASURE_1> <CUTOFF_1>; ... ;<MEASURE_n> <CUTOFF_n>= (etc),
+where each \lstinline=MEASURE_N= is one of the options below (note
+these corespond to all the rank methods accepted under the rank setting
+except for \lstinline=random= which does not make sense as a threshold
+measure). Spaces delimit corresponding measures and their values, while
+semi-colons (optionally with subsequent spaces) delimit each measure-value
```

Comment:
Out of interest, how does this setting play with multiline strings?

```ini
[suite servers]
    [[run host select]]
        thresholds = """
            load: 15 1
            memory: 200000
        """
        
```

Reply:
``HostAppointer.parse_thresholds`` relies on splitting threshold-cutoff pairs by semi-colon, so that syntax would not work, but I can tweak the code to support it if you think it would be worthwhile (is this a common syntax for compound global config strings)?

Reply:
The `"""` syntax should work for all strings? I'm guessing the previous example would have worked if the thresholds were semi-colon delimited.

Reply:
Yes the semi-colons are the crucial part. Multi-line strings would not be a problem.